onnxruntime/test/providers/cpu/math/cumsum_test.cc

```diff
@@ -14,6 +16,15 @@ TEST(CumSumTest, _1DTest) {
   test.AddOutput<float>("y", {5}, {1., 3., 6., 10., 15.});
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
 }
+TEST(CumSumTest, _1DTestFloat16) {
+  if (DefaultCudaExecutionProvider().get() != nullptr) {
```

Comment:
nit: Do we really need this check given that CPU EP has been added to the exclusion list for this test? On a related note, do you want to register the CPU opset-14 kernel for this op or do you think we can do it later ?

Reply:
I think we can skip float16 for CPU right now. This check is to avoid test failure (like no available EP for the test) for CPU build.

Reply:
It is true that we can skip float16 support for CPU right now - but we can register a kernel for opset 14 CumSum for the CPU EP (supporting already supported types in the opset 13 kernel) and that way you can remove the if here, because it should still use the `float16 -> float ->CumSum float kernel-> float16` casting logic and run the test for the CPU EP. Anyway, we can do that in a separate change.